webui/react/src/pages/Cluster/ClusterOverview.tsx

```diff
@@ -41,7 +41,10 @@ const defaultSorter = { descend: false, key: 'name' };
  * @param pool resource pool
  */
 export const maxPoolSlotCapacity = (pool: ResourcePool): number => {
-  return pool.maxAgents * (pool.slotsPerAgent ?? 0);
+  if (pool.maxAgents) return pool.maxAgents * (pool.slotsPerAgent ?? 0);
+  // on-premise deployments don't have dynamic agents and we don't know how many
+  // agents might connect.
+  return pool.slotsAvailable;
```

Comment:
@ioga does this look right to you?

Reply:
seems like it'll work.

on a related note, the `pool.slotsPerAgent ?? 0` has always seemed "wrong" to me because the backend still returns `-1` for static pools. however `maxAgents` for static pools is always 0, which saves us from any negative numbers appearing here. So, probably not a big deal. But it'd be great if some day backend uses "null" value for unknown `slotsPerAgent`.

Reply:
it doesn't hurt to take slotsPerAgent < 1 out of consideration here. since now we're reusing this bit it should be a one-line change. I'll update it